File: node/samples/formCardSample.ts
import {
  AdaptiveCardBuilder,
  TextSize,
  TextWeight,
  TextInputStyle,
  ActionStyle,
  ChoiceInputStyle,
} from 'fluent-cards';

/** Creates a contact form card. */
export function createContactForm() {
  return AdaptiveCardBuilder.create()
    .withVersion('1.5')
    .addTextBlock((tb) =>
      tb
        .withText('Contact Us')
        .withSize(TextSize.Large)
        .withWeight(TextWeight.Bolder),
    )
    .addInputText((i) =>
      i
        .withId('name')
        .withLabel('Name')
        .withPlaceholder('Enter your name')
        .withIsRequired()
        .withErrorMessage('Name is required'),
    )
    .addInputText((i) =>
      i
        .withId('email')
        .withLabel('Email')
        .withPlaceholder('Enter your email')
        .withStyle(TextInputStyle.Email)
        .withIsRequired(),
    )
    .addInputText((i) =>
      i
        .withId('message')
        .withLabel('Message')
        .withPlaceholder('How can we help?')
        .withIsMultiline()
        .withMaxLength(500),
    )
    .addAction((a) => a.submit('Send Message').withStyle(ActionStyle.Positive))
    .build();
}

/** Creates a survey form card. */
export function createSurveyForm() {
  return AdaptiveCardBuilder.create()
    .withVersion('1.5')
    .addTextBlock((tb) =>
      tb
        .withText('Customer Satisfaction Survey')
        .withSize(TextSize.Large)
        .withWeight(TextWeight.Bolder),
    )
    .addInputChoiceSet((i) =>
      i
        .withId('satisfaction')
        .withLabel('How satisfied are you?')
        .addChoice('Very Satisfied', '5')
        .addChoice('Satisfied', '4')
        .addChoice('Neutral', '3')
        .addChoice('Dissatisfied', '2')
        .addChoice('Very Dissatisfied', '1')
        .withIsRequired(),
    )
    .addInputText((i) =>
      i
        .withId('feedback')
        .withLabel('Additional Feedback')
        .withPlaceholder('Tell us more...')
        .withIsMultiline(),
    )
    .addAction((a) => a.submit('Submit Survey').withStyle(ActionStyle.Positive))
    .build();
}

/** Creates a registration form card. */
export function createRegistrationForm() {
  return AdaptiveCardBuilder.create()
    .withVersion('1.5')
    .addTextBlock((tb) =>
      tb
        .withText('Event Registration')
        .withSize(TextSize.Large)
        .withWeight(TextWeight.Bolder),
    )
    .addInputText((i) =>
      i.withId('fullName').withLabel('Full Name').withIsRequired(),
    )
    .addInputText((i) =>
      i
        .withId('email')
        .withLabel('Email Address')
        .withStyle(TextInputStyle.Email)
        .withIsRequired(),
    )
    .addInputDate((i) => i.withId('eventDate').withLabel('Event Date'))
    .addInputToggle((i) =>
      i
        .withId('newsletter')
        .withTitle('Subscribe to newsletter')
        .withValue('true'),
    )
    .addAction((a) => a.submit('Register').withStyle(ActionStyle.Positive))
    .build();
}

File: python/samples/form_card_sample.py
from fluent_cards import (
    AdaptiveCardBuilder,
    TextSize,
    TextWeight,
    TextInputStyle,
    ActionStyle,
    ChoiceInputStyle,
)


def create_contact_form():
    """Creates a contact form card."""
    return (
        AdaptiveCardBuilder.create()
        .with_version('1.5')
        .add_text_block(
            lambda tb: tb.with_text('Contact Us')
            .with_size(TextSize.Large)
            .with_weight(TextWeight.Bolder)
        )
        .add_input_text(
            lambda i: i.with_id('name')
            .with_label('Name')
            .with_placeholder('Enter your name')
            .with_is_required()
            .with_error_message('Name is required')
        )
        .add_input_text(
            lambda i: i.with_id('email')
            .with_label('Email')
            .with_placeholder('Enter your email')
            .with_style(TextInputStyle.Email)
            .with_is_required()
        )
        .add_input_text(
            lambda i: i.with_id('message')
            .with_label('Message')
            .with_placeholder('How can we help?')
            .with_is_multiline()
            .with_max_length(500)
        )
        .add_action(
            lambda a: a.submit('Send Message').with_style(ActionStyle.Positive)
        )
        .build()
    )


def create_survey_form():
    """Creates a survey form card."""
    return (
        AdaptiveCardBuilder.create()
        .with_version('1.5')
        .add_text_block(
            lambda tb: tb.with_text('Customer Satisfaction Survey')
            .with_size(TextSize.Large)
            .with_weight(TextWeight.Bolder)
        )
        .add_input_choice_set(
            lambda i: i.with_id('satisfaction')
            .with_label('How satisfied are you?')
            .add_choice('Very Satisfied', '5')
            .add_choice('Satisfied', '4')
            .add_choice('Neutral', '3')
            .add_choice('Dissatisfied', '2')
            .add_choice('Very Dissatisfied', '1')
            .with_is_required()
        )
        .add_input_text(
            lambda i: i.with_id('feedback')
            .with_label('Additional Feedback')
            .with_placeholder('Tell us more...')
            .with_is_multiline()
        )
        .add_action(
            lambda a: a.submit('Submit Survey').with_style(ActionStyle.Positive)
        )
        .build()
    )


def create_registration_form():
    """Creates a registration form card."""
    return (
        AdaptiveCardBuilder.create()
        .with_version('1.5')
        .add_text_block(
            lambda tb: tb.with_text('Event Registration')
            .with_size(TextSize.Large)
            .with_weight(TextWeight.Bolder)
        )
        .add_input_text(
            lambda i: i.with_id('fullName').with_label('Full Name').with_is_required()
        )
        .add_input_text(
            lambda i: i.with_id('email')
            .with_label('Email Address')
            .with_style(TextInputStyle.Email)
            .with_is_required()
        )
        .add_input_date(
            lambda i: i.with_id('eventDate').with_label('Event Date')
        )
        .add_input_toggle(
            lambda i: i.with_id('newsletter')
            .with_title('Subscribe to newsletter')
            .with_value('true')
        )
        .add_action(
            lambda a: a.submit('Register').with_style(ActionStyle.Positive)
        )
        .build()
    )
